File: frontend/types/attorney.ts
/**
 * 노무사 마켓플레이스 타입 정의
 */

// 케이스 유형
export type CaseType = 'dismissal' | 'wage' | 'leave' | 'industrial_accident' | 'harassment' | 'other';

// 긴급도
export type Urgency = 'low' | 'medium' | 'high' | 'emergency';

// 상담 방식
export type ConsultationType = 'video' | 'phone' | 'visit';

// 케이스 상태
export type CaseStatus = 'pending' | 'accepted' | 'in_progress' | 'completed' | 'cancelled';

// 라벨 매핑
export const CASE_TYPE_LABELS: Record<CaseType, string> = {
  dismissal: '해고/퇴직',
  wage: '임금/수당',
  leave: '휴가/휴직',
  industrial_accident: '산업재해',
  harassment: '직장 내 괴롭힘',
  other: '기타',
};

export const URGENCY_LABELS: Record<Urgency, string> = {
  low: '낮음',
  medium: '보통',
  high: '높음',
  emergency: '긴급',
};

export const CONSULTATION_TYPE_LABELS: Record<ConsultationType, string> = {
  video: '화상 상담',
  phone: '전화 상담',
  visit: '방문 상담',
};

export const CASE_STATUS_LABELS: Record<CaseStatus, string> = {
  pending: '대기중',
  accepted: '수락됨',
  in_progress: '진행중',
  completed: '완료',
  cancelled: '취소됨',
};

export const CASE_STATUS_COLORS: Record<CaseStatus, string> = {
  pending: 'bg-yellow-100 text-yellow-800',
  accepted: 'bg-blue-100 text-blue-800',
  in_progress: 'bg-indigo-100 text-indigo-800',
  completed: 'bg-green-100 text-green-800',
  cancelled: 'bg-slate-100 text-slate-500',
};

// 노무사 정보
export interface Attorney {
  id: string;
  name: string;
  firm_name: string;
  specialties: string[];
  regions: string[];
  consultation_fee: number;
  experience_years: number;
  rating: number;
  review_count: number;
  response_rate: number;
  bio: string | null;
  profile_image_url: string | null;
  verified: boolean;
}

// 노무사 상세 (리뷰 포함)
export interface AttorneyDetail {
  attorney: Attorney;
  recent_reviews: Review[];
}

// 리뷰
export interface Review {
  id: string;
  rating: number;
  comment: string | null;
  user_name: string;
  created_at: string;
}

// 케이스
export interface AttorneyCase {
  case_id: string;
  attorney_id: string;
  attorney_name: string;
  case_type: CaseType;
  urgency: Urgency;
  status: CaseStatus;
  consultation_type: string | null;
  description: string | null;
  consultation_fee: number;
  created_at: string;
}

// 상담 신청 요청
export interface CreateCaseRequest {
  attorney_id: string;
  case_type: CaseType;
  urgency: Urgency;
  consultation_type?: ConsultationType;
  description?: string;
}

// 상담 신청 결과
export interface CreateCaseResult {
  case_id: string;
  status: string;
  attorney_name: string;
  case_summary: string | null;
  consultation_fee: number;
}

// 리뷰 작성 요청
export interface CreateReviewRequest {
  rating: number;
  comment?: string;
}

// 노무사 목록 응답
export interface AttorneyListData {
  attorneys: Attorney[];
  total_count: number;
}

// 케이스 목록 응답
export interface CaseListData {
  cases: AttorneyCase[];
  total_count: number;
}

// 리뷰 목록 응답
export interface ReviewListData {
  reviews: Review[];
  total_count: number;
}

File: backend/app/schemas/attorney.py
# 노무사 마켓플레이스 스키마
from datetime import datetime
from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID


# ===== 노무사 프로필 =====
class AttorneyResponse(BaseModel):
    """노무사 응답"""
    id: UUID
    name: str
    firm_name: Optional[str]
    specialties: List[str]
    regions: List[str]
    consultation_fee: int
    experience_years: int
    rating: float
    review_count: int
    response_rate: int
    bio: Optional[str]
    profile_image_url: Optional[str]
    verified: bool


class AttorneyDetailResponse(BaseModel):
    """노무사 상세 응답"""
    attorney: AttorneyResponse
    recent_reviews: List["ReviewResponse"]


# ===== 상담 케이스 =====
class CreateCaseRequest(BaseModel):
    """상담 신청 요청"""
    attorney_id: UUID
    company_id: Optional[UUID] = None
    chat_session_id: Optional[UUID] = None
    case_type: str = Field(..., pattern=r"^(dismissal|wage|leave|industrial_accident|harassment|other)$")
    urgency: str = Field(..., pattern=r"^(low|medium|high|emergency)$")
    consultation_type: str = Field(default="video", pattern=r"^(video|phone|visit)$")
    preferred_schedule: Optional[List[str]] = None
    description: Optional[str] = None


class CaseResponse(BaseModel):
    """케이스 응답"""
    id: UUID
    attorney_id: UUID
    attorney_name: str
    case_summary: str
    case_type: str
    urgency: str
    status: str
    consultation_type: str
    consultation_fee: int
    scheduled_at: Optional[datetime]
    fee_paid: bool
    completed_at: Optional[datetime]
    created_at: datetime


class CaseCreateResult(BaseModel):
    """케이스 생성 결과"""
    case_id: UUID
    case_summary: str
    status: str
    consultation_fee: int


# ===== 리뷰 =====
class CreateReviewRequest(BaseModel):
    """리뷰 작성 요청"""
    rating: int = Field(..., ge=1, le=5)
    comment: Optional[str] = None


class ReviewResponse(BaseModel):
    """리뷰 응답"""
    id: UUID
    rating: int
    comment: Optional[str]
    user_name: str
    created_at: datetime


class ReviewCreateResult(BaseModel):
    """리뷰 생성 결과"""
    review_id: UUID


# ===== 페이지네이션 =====
class AttorneyListResponse(BaseModel):
    """노무사 목록 응답"""
    attorneys: List[AttorneyResponse]
    total_count: int


class CaseListResponse(BaseModel):
    """케이스 목록 응답"""
    cases: List[CaseResponse]
    total_count: int


class ReviewListResponse(BaseModel):
    """리뷰 목록 응답"""
    reviews: List[ReviewResponse]
    total_count: int


# Forward ref 해결
AttorneyDetailResponse.model_rebuild()
